compiler/rustc_mir/src/borrow_check/diagnostics/conflict_errors.rs

```diff
@@ -1564,7 +1577,22 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
             }));
         }
 
-        result
+        // Check if we can reach these reinits from a move location.
+        let mut maybe_reinitialized = false;
+        let mut visited = FxHashSet::default();
+        let mut stack = reinits;
```

Comment:
At first we just collect the potential reinits we found and then use those to initialize the stack for the search. I thought this would be less confusing. Alternatively we could rename reinits to something like stack_reinits and remove the rename.

Reply:
The rename makes sense to me.